src/Clapi/Types/Path.hs

```diff
@@ -64,8 +64,14 @@ pattern seg :</ path <- (splitHead -> Just (seg, path)) where
     seg :</ path = Path $ seg : unPath path
 
 splitTail :: Path -> Maybe (Path, Seg)
-splitTail (Path []) = Nothing
-splitTail (Path segs) = Just (Path $ init segs, last segs)
+splitTail (Path path) = case path of
+    (y : xs) -> (\(s, ps) -> Just (Path ps, s)) $ go y xs
```

Comment:
Not that this is an issue, but could have been written as
```
let (s, ps) = go y xs in Just (Path ps, s)
```

Reply:
True, this one fell out from an adaptation of a previous draft that had more functors (and thus a `<$>`) which wasn't so friendly to let. Would be clearer in your suggested formulation.